src/Symfony/Component/Form/Test/FormIntegrationTestCase.php

```diff
@@ -29,11 +29,17 @@ protected function setUp()
     {
         $this->factory = Forms::createFormFactoryBuilder()
             ->addExtensions($this->getExtensions())
+            ->addTypeExtensions($this->getTypeExtensions())
             ->getFormFactory();
     }
 
     protected function getExtensions()
     {
         return array();
     }
+
+    protected function getTypeExtensions()
```

Comment:
If this line were too be added in favor of:
```php
use Symfony\Component\Form\PreloadedExtension;

// ...

protected function getExtension()
{
    return array(new PreloadedExtension(
        array(new SomeType($this->getDependencyMock())),
        array(new SomeTypeExtension(new SomeService()),
        array(
            new SomeGuesser($this->getDependencyMock2()),
            new SomeOtherGuesser()),
    ));
}
```
We should also add two other protected functions for types and guessers, and add `addTypeGuessers` and `addTypes` call above as well to be consistent.

I would not vote against it, but I would vote for a better doc.